framework/src/Volo.Abp.Emailing/Volo/Abp/Emailing/EmailSenderBase.cs

```diff
@@ -137,52 +87,139 @@ public virtual async Task QueueAsync(string from, string to, string subject, str
+            : new MailMessage(from, to, subject, body) { IsBodyHtml = isBodyHtml };
+
+        if (additionalEmailSendingArgs != null)
+        {
+            if (additionalEmailSendingArgs.Attachments != null)
+            {
+                foreach (var attachment in additionalEmailSendingArgs.Attachments.Where(x => x.File != null))
+                {
+                    var fileStream = new MemoryStream(attachment.File!);
+                    _ = fileStream.Seek(0, SeekOrigin.Begin);
+                    message.Attachments.Add(new Attachment(fileStream, attachment.Name));
+                }
+            }
+
+            if (additionalEmailSendingArgs.CC != null)
+            {
+                foreach (var cc in additionalEmailSendingArgs.CC)
+                {
+                    message.CC.Add(cc);
+                }
+            }
+        }
+
+        return message;
+    }
 
     /// <summary>
-    /// Normalizes given email.
-    /// Fills <see cref="MailMessage.From"/> if it's not filled before.
-    /// Sets encodings to UTF8 if they are not set before.
+    /// Normalizes given email. Fills <see cref="MailMessage.From"/> if it's not filled before. Sets
+    /// encodings to UTF8 if they are not set before.
     /// </summary>
     /// <param name="mail">Mail to be normalized</param>
     protected virtual async Task NormalizeMailAsync(MailMessage mail)
     {
         if (mail.From == null || mail.From.Address.IsNullOrEmpty())
         {
-            mail.From = new MailAddress(
-                await Configuration.GetDefaultFromAddressAsync(),
-                await Configuration.GetDefaultFromDisplayNameAsync(),
-                Encoding.UTF8
-                );
+            mail.From = new MailAddress(await Configuration.GetDefaultFromAddressAsync(),
+                                        await Configuration.GetDefaultFromDisplayNameAsync(),
+                                        Encoding.UTF8);
         }
 
-        if (mail.HeadersEncoding == null)
-        {
-            mail.HeadersEncoding = Encoding.UTF8;
-        }
+        mail.HeadersEncoding ??= Encoding.UTF8;
 
-        if (mail.SubjectEncoding == null)
-        {
-            mail.SubjectEncoding = Encoding.UTF8;
-        }
+        mail.SubjectEncoding ??= Encoding.UTF8;
+
+        mail.BodyEncoding ??= Encoding.UTF8;
+    }
+
+    /// <summary>
+    /// Should implement this method to send email in derived classes.
+    /// </summary>
+    /// <param name="mail">Mail to be sent</param>
+    protected abstract Task SendEmailAsync(MailMessage mail);
 
-        if (mail.BodyEncoding == null)
+    private static void NormalizeMailForBase64Image(MailMessage mail)
+    {
+        var htmlBody = mail.Body;
+
+        // Find all base64 image tags in the html message.
+        var base64ImageTags = Regex.Matches(htmlBody, @"<img src=""data:image/(png|jpeg|gif);base64,([^""]+)""[^>]*>");
```

Comment:
This regex doesn't cover all variants of a base64 image element.
Here's a better regex that finds all possible `src` with the base64 format:

```
src(\s*)=(\s*)("|')(\s*)data(\s*):(\s*)image/(\s*)(png|jpeg|gif);(\s*)base64,([^""]+).("|')
```

 
![image](https://github.com/abpframework/abp/assets/9526587/66931233-da18-4a5c-b400-178c651a5c1d)

Reply:
@ebicoglu 
I think it would be better to use the following regex pattern:

`src\s*=\s*(""|')\s*data\s*:\s*(?<mediaType>[\w/\-\.]+);(?<encoding>\w+),(?<data>.*)(""|')`

Because it covers all tags that use base64, such as fonts, <video>, <audio>, and more, not just limited to images.